### ISSUE TITLE
yolo26*-seg end2end duplicate detections

### ISSUE THREAD

**Issue by dk-teknologisk-mlnn** [ultralytics/ultralytics#23685]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi,

Im getting duplicate detections ( ~100% overlap) if I use the new yolo26 segmentation models. 
if I run with end2end=false, nms=true I get good results.

 _results = self.model(images_list, verbose=False, retina_masks=True, conf=conf_threshold, iou=0.5, max_det=300, nms=not self.end2end, end2end=self.end2end)_

I wonder if its a known drawback? or it is a sign I need to train more?

The model has it easy, just large high contrast circles for now. it is not splitting them up, its the full circle I get multiple detections on when self.end2end is true .

 Ultralytics  8.4.14 python 3.11, Windows 11, pytorch 2.3 and 2.10, cuda 11.8 and 12.6

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dk-teknologisk-mlnn, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unblocked quickly—an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report (duplicate detections with `end2end=True` on YOLO26 segmentation), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can reproduce and debug the behavior. 🙏  
To help us triage faster, please include in the MRE:
-Your exact `ultralytics` version and install method
-A complete runnable snippet that loads your model and runs inference with the exact arguments you shared (`end2end`, `nms`, `conf`, `iou`, `max_det`, `retina_masks`)
-A small set of sample images (or a way to generate them) that shows the near-100% overlap duplicates
-The full console output (including any warnings) and a brief note on expected vs actual results

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 🧠

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ulralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
If it's a 100% overlap and different class, you can remove those by running with `agnostic_nms=True` and without having to disable `end2end`

**Comment by dk-teknologisk-mlnn** [comment]
the model has 1 class. so its same class.

**Comment by Y-T-G** [comment]
How many images did you use for training?

**Comment by dk-teknologisk-mlnn** [comment]
20 . not many, thats why I made this a question ( not a bug) if its to be expected that end2end wont perform as well as the classic nms method, when its not trained much ( 20 images 100 epochs?
end2end=false works flawless btw.

**Comment by Y-T-G** [comment]
Yes, it requires a larger dataset for NMS-free mode
